Test/Tests/RealtimeClientConnectionTests.swift

```diff
@@ -2444,15 +2529,18 @@ class RealtimeClientConnectionTests: XCTestCase {
                 let connectedPM = transport.protocolMessagesReceived.filter { $0.action == .connected }[0]
                 XCTAssertEqual(connectedPM.connectionId, expectedConnectionId)
                 XCTAssertNil(stateChange.reason)
+                // Attach should be 2 here, as one above and one reattach
```

Comment:
I wonder if these tests with this "as one above and one reattach" pattern would be clearer if you first wait for the explicit attach to complete, and verify that 1 `ATTACH` has been sent. What do you think?

Reply:
It's a bug f06ff023cb7f6224086779a75c6b2a03cb842eed

Reply:
Sorry, could you explain please? I was asking about the tests but the commit you've linked to doesn't have any test changes.

Would you also be able to explain what's going on in the commit that you've linked to? I am struggling to understand what the code that you've removed was trying to do, which makes me a bit nervous about removing it. It looks like it was added in ba55f9ec0b34e92d3ce1e0d30f7447a0ce8441e8 but honestly after having spent most of the day reviewing this PR my brain is in no state to try and understand what that commit was trying to do 🤯

Reply:
My initial comment `// Attach should be 2 here, as one above and one reattach` was misleading (I've made erroneous conclusion about what gives me the second `ATTACH` in sent messages). I've understood it when was trying to address you comment about other tests with "as one above and one reattach" pattern. The count wasn't always 2 for sent messages in other tests, and after some debugging I've located this place where the second `ATTACH` message was generated. I believe Ricardo forgot to remove this in 784da01, because earlier re-attach would not happen,  because `RTL4i` block was in the `else if (![self shouldQueueEvents])` part of `performTransitionWithStateChange:` and re-attaching was made upon emitting `[_connectedEventEmitter emit:nil with:nil]` in `case ARTRealtimeConnected:`. Once Recardo moved it to `if ([self shouldSendEvents])` re-attaching from emitter should have been be deleted.

Reply:
Sorry, but there's a lot going on here and I'm struggling to juggle it all in my brain, including your explanation. But, it sounds like this change is unrelated to this PR. Maybe it would help me to understand if you opened a separate PR for removing this code that you believe shouldn't be there, explaining why you believe it was originally added, and why you believe it should have been removed earlier?

Reply:
https://github.com/ably/ably-cocoa/pull/1822